backend/src/connector/mod.rs

```diff
@@ -1 +1,267 @@
+    /// Creates new [`BobClient`] from [`BobConnectionData`]
+    ///
+    /// # Errors
+    /// The function will fail if a hostname isn't a valid url or the client couldn't establish
+    /// connection for the BOB cluster
+    pub async fn try_new(
+        bob_data: BobConnectionData,
+        timeout: RequestTimeout,
+    ) -> Result<Self, ClientError> {
+        let auth = if let Some(creds) = bob_data.credentials {
+            Some(Authorization::basic(&creds.login, &creds.password))
+        } else {
+            None
+        };
+        let hostname = bob_data.hostname.clone();
+
+        let context: ClientContext = hlist![timeout, auth, XSpanIdString::default()];
+        let client = Box::new(
+            Client::try_new_http(&hostname.to_string()).change_context(ClientError::InitClient)?,
+        );
+        let nodes = client
+            .with_context(context.clone())
+            .get_nodes()
+            .await
+            .change_context(ClientError::InitClient)?;
+        let api::GetNodesResponse::AJSONArrayOfNodesInfoAndVdisksOnThem(nodes) = nodes else {
+            Err(APIError::InvalidStatusCode(StatusCode::FORBIDDEN))
+                .change_context(ClientError::PermissionDenied)?
+        };
+
+        let cluster: HashMap<NodeName, Arc<ApiInterface>> = nodes
+            .iter()
+            .map(|node| {
+                (
+                    &node.name,
+                    Self::change_port(&node.address, &bob_data.hostname),
+                )
+            })
+            .filter_map(|(name, hostname)| {
+                if hostname.is_err() {
+                    tracing::warn!("couldn't change port for {name}. Client won't be created");
+                }
+                hostname
+                    .ok()
+                    .map(|hostname| (name, Client::try_new_http(&hostname.to_string())))
+            })
+            .filter_map(|(name, client)| {
+                if client.is_err() {
+                    tracing::warn!("couldn't create client for {hostname}");
+                }
+                client.ok().map(|client| {
+                    (
+                        name.clone(),
+                        Arc::new(client.with_context(context.clone())) as Arc<ApiInterface>,
+                    )
+                })
+            })
+            .collect();
+        let client = Box::new(
+            Client::try_new_http(&hostname.to_string()).change_context(ClientError::InitClient)?,
+        );
+        Ok(Self {
+            hostname: bob_data.hostname,
+            main: Arc::new(client.with_context(context)),
+            cluster,
+        })
+    }
+
+    /// Probes connection to the Bob cluster
+    ///
+    /// Returns `StatusCode::OK` on success
+    ///
+    /// # Errors
+    ///
+    /// The function fails if there was an error during creation of request
+    /// It shouldn't happen on normal behaviour
+    ///
+    pub async fn probe(&self) -> Result<StatusCode, ClientError> {
```

Comment:
Maybe call it `proble_main` so the following function can be called `probe_secondary`